src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/IconCacheService.cs

```diff
@@ -20,15 +20,23 @@ public sealed class IconCacheService(DispatcherQueue dispatcherQueue)
 
     private async Task<IconSource?> IconToSource(IconDataType icon)
     {
-        if (!string.IsNullOrEmpty(icon.Icon))
+        // bodgy: apparently IconDataType, despite being a struct, doesn't get
+        // MarshalByValue'd into our process. What's even the point then?
```

Comment:
😲

Reply:
Note to self: make an IconViewModel ala the way I had it in #218, as to copy the data locally. 

Though, I guess the `IRandomAccessStreamReference` can't actually be copied locally - that one will just have to be accessed carefully when we want it.